Assets/Scripts/Player/PlayerMain.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections;
+using UnityEngine;
+using Cinemachine;
+
+using Actor;
+
+namespace Player {
+
+    public class PlayerMain : MonoBehaviour, iActor {
+        [SerializeField] CinemachineImpulseSource screenShakeOnDamage;
+        [SerializeField] CinemachineImpulseSource screenShakeOnDeath;
+
+        // props
+        System.Guid guid = new System.Guid(); // this is the unique ID used for comparing enemies, bosses, pickups, destructibles etc.
+
+        // cached
+        Health health;
+
+        void OnEnable() {
+            if (health != null) {
+                health.OnDamageTaken.Subscribe(OnDamageTaken);
+                health.OnDeath.Subscribe(OnDeath);
+            }
+
+            // TODO: REMOVE
+            controller.OnFirePress.Subscribe(OnFirePress);
+            controller.OnMeleePress.Subscribe(OnMeleePress);
+        }
+
+        void OnDisable() {
+            if (health != null) {
+                health.OnDamageTaken.Unsubscribe(OnDamageTaken);
+                health.OnDeath.Unsubscribe(OnDeath);
+            }
+
+            // TODO: REMOVE
+            controller.OnFirePress.Unsubscribe(OnFirePress);
+            controller.OnMeleePress.Unsubscribe(OnMeleePress);
+        }
+
+        void Awake() {
+            health = GetComponent<Health>();
+
+            // TODO: REMOVE
+            controller = GetComponent<PlayerController>();
```

Comment:
This makes sense to place in a PlayerWeapons class or something like that, hence the TODO: REMOVE here